src/mscorlib/src/System/Number.cs

```diff
@@ -685,6 +685,14 @@ internal unsafe struct NumberBuffer {
 
         [System.Security.SecuritySafeCritical]  // auto-generated
         internal unsafe static Int32 ParseInt32(String s, NumberStyles style, NumberFormatInfo info) {
+            if (style == NumberStyles.Integer && info == NumberFormatInfo.InvariantInfo) {
```

Comment:
The current code would be called in this situation
`int.Parse("123", NumberStyles.Integer, NumberFormatInfo.InvariantInfo)`, but not `int.Parse("123")`
I agree with you that the benefits would be considerably more widespread if we dropped the requirement for a specific numberformatinfo, as NumberStyles.Integer is the same across cultures